suprb/optimizer/solution/ga/crossover.py

```diff
@@ -8,27 +8,49 @@
 
 
 class SolutionCrossover(BaseComponent, metaclass=ABCMeta):
-
-    def __init__(self, crossover_rate: float = 0.9):
-        self.crossover_rate = crossover_rate
-
-    def __call__(self, A: Solution, B: Solution, random_state: RandomState) -> Solution:
-        if random_state.random() < self.crossover_rate:
+    def __call__(self, A: Solution, B: Solution, crossover_rate, random_state: RandomState) -> Solution:
+        if random_state.random() < crossover_rate:
             return self._crossover(A=A, B=B, random_state=random_state)
-        else:
-            # Just return the primary parent
-            return A
+
+        return A
```

Comment:
not sure if the else wouldn't improve readability. what is the current guideline on this?

Reply:
I think the current guidelines will have us save the result in a variable and then return it (to not have multiple return statements in the same function.